atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/qos/ThriftQueryWeighersTest.java

```diff
@@ -77,26 +97,54 @@ public void multigetSliceWeigherReturnsCorrectNumRows() {
 
     @Test
     public void rangeSlicesWeigherEstimatesNumberOfBytesBasedOnNumberOfRows() {
-        long numBytesWithOneRow = ThriftQueryWeighers.getRangeSlices(new KeyRange(1)).estimate().numBytes();
-        long numBytesWithTwoRows = ThriftQueryWeighers.getRangeSlices(new KeyRange(2)).estimate().numBytes();
+        assertThatEstimatesAreCorrectForRangeSlices(false);
+    }
 
-        assertThat(numBytesWithOneRow).isGreaterThan(0L);
-        assertThat(numBytesWithTwoRows).isEqualTo(numBytesWithOneRow * 2);
+    @Test
+    public void rangeSlicesWeigherWithZeroEstimateEstimatesZeroNumberOfBytes() {
+        assertThatEstimatesAreCorrectForRangeSlices(true);
+    }
+
+    private void assertThatEstimatesAreCorrectForRangeSlices(boolean zeroEstimate) {
+        long numBytesWithOneRow = ThriftQueryWeighers.getRangeSlices(new KeyRange(1), zeroEstimate)
+                .estimate().numBytes();
+        long numBytesWithTwoRows = ThriftQueryWeighers.getRangeSlices(new KeyRange(2), zeroEstimate)
+                .estimate().numBytes();
+
+        if (zeroEstimate) {
+            assertThat(numBytesWithOneRow).isEqualTo(0L);
+        } else {
+            assertThat(numBytesWithOneRow).isGreaterThan(0L);
+        }
+        assertThat(numBytesWithTwoRows).isEqualTo(zeroEstimate ? 0L : numBytesWithOneRow * 2);
```

Comment:
nit: this block looks similar to the one at the end of `assertThatEstimatesAreCorrect` - we might want to extract a method?

Reply:
Done.